external/mmdetection/detection_tasks/apis/detection/openvino_task.py

```diff
@@ -398,27 +375,26 @@ def predict_tile(
             if not detection_task:
                 metadata["resize_mask"] = False
             raw_predictions = self.model.infer_sync(tile_dict)
-            detections = self.model.postprocess(raw_predictions, metadata)
-            score, label, box, mask = self.process_tile_prediction(detections, offset)
-            scores.extend(score)
-            labels.extend(label)
-            boxes.extend(box)
+            output = self.model.postprocess(raw_predictions, metadata)
+            score, label, box, mask = self.process_tile_prediction(output, offset)
+            scores = np.append(scores, score, axis=0)
+            labels = np.append(labels, label, axis=0)
+            boxes = np.append(boxes, box, axis=0)
             masks.extend(mask)
 
-        scores = np.asarray(scores, dtype=np.float32)
-        labels = np.asarray(labels, dtype=np.uint32)
-        boxes = np.asarray(boxes, dtype=np.float32)
         assert len(scores) == len(labels) == len(boxes) == len(masks)
-        _, keep = multiclass_nms(scores, labels, boxes, max_num=max_number)
-        boxes = boxes[keep]
-        labels = labels[keep]
-        scores = scores[keep]
-        masks = [masks[keep_idx] for keep_idx in keep]
-
-        for i in range(len(boxes)):
-            masks[i] = self.model._segm_postprocess(
-                boxes[i], masks[i], *original_shape[:-1]
-            )
+        if np.prod(scores):
```

Comment:
AFAIK, `np.prod(x)` returns 1 when `x=np.array([])` . Is it okay? How about using `.shape` if you want to check whether `scores` is empty or not